stonesoup/sampler/particle.py

```diff
@@ -0,0 +1,149 @@
+            samples = samples.T
+
+        particles = ParticleState(state_vector=StateVectors(samples),
+                                  weight=np.array([1 / nsamples] * nsamples),
+                                  timestamp=timestamp)
+
+        return particles
+
+
+class GaussianDetectionParticleSampler(ParticleSampler):
+
+    distribution_func = None
+    params = None
+    ndim_state = None
+
+    nsamples: int = Property(
+        default=1,
+        doc="Number of samples to return")
+
+    def sample(self, detections):
+        """Samples from a gaussian mixture around detections
+
+        Parameters
+        ----------
+        detections : :class:`~.Detection` or set of :class:`~.Detection`
+            Detections forming the distribution to sample from.
+        Returns
+        -------
+        : :class:`~.ParticleState`
+            :class:`~.ParticleState` of samples."""
+
+        dist_mean = []
+        dist_covar = []
+        num_det = len(detections)
+        timestamp = next(iter(detections)).timestamp
+        for detection in detections:
+            ndim_state = detection.measurement_model.ndim_state
+            ndim_meas = detection.measurement_model.ndim
+            if isinstance(detection.measurement_model, LinearModel):
+                if ndim_state > ndim_meas:
+                    mapping = detection.measurement_model.mapping
+                    mapping_matrix = np.zeros((ndim_state, ndim_meas))
+                    mapping_index = np.linspace(0, len(mapping)-1, ndim_meas, dtype=int)
+                    mapping_matrix[mapping, mapping_index] \
+                        = 1
+                    dist_mean.append(mapping_matrix @ detection.state_vector)
+                    dist_covar.append(mapping_matrix @
+                                      detection.measurement_model.noise_covar @
+                                      mapping_matrix.T)
+                else:
+                    dist_mean.append(detection.state_vector)
+                    dist_covar.append(detection.measurement_model.noise_covar)
+            else:
+                tmp_mean = detection.measurement_model.inverse_function(detection)
+                jac = jacobian(detection.measurement_model.inverse_function, detection)
+                tmp_covar = jac @ detection.measurement_model.noise_covar @ jac.T
+                dist_mean.append(tmp_mean)
+                dist_covar.append(tmp_covar)
+
+        weights = self.get_weight(num_det)
+
+        params = {'means': dist_mean,
+                  'covars': dist_covar,
+                  'weights': weights,
+                  'size': self.nsamples}
+        samples = super().sample(distribution_func=gm_sample,
+                                 params=params,
+                                 ndim_state=ndim_state,
+                                 timestamp=timestamp)
+
+        return samples
+
+    @staticmethod
+    def get_weight(num_detections):
```

Comment:
Is this necessary? it's only used once.

Reply:
I added the `get_weight` method to make it easier for others to insert their code if doing something novel with the weighting between the detection components.